modules/dnn/src/webgpu/dawnWrapperTest/op_softmax_test.cpp

```diff
@@ -0,0 +1,31 @@
+#include <stdio.h>
+#include <opencv2/opencv.hpp>
+#include "../include/op_softmax.hpp"
+#include "../src/context.hpp"
+#include "../include/wgpucom.hpp"
+#include <unistd.h>
+#include "../src/common.hpp"
+
+using namespace cv::dnn;
+int main(int argc, char** argv )
+{
+    webgpu::wDevice = std::make_shared<wgpu::Device>(webgpu::createCppDawnDevice());
+    webgpu::wQueue = std::make_shared<wgpu::Queue>(webgpu::wDevice->GetDefaultQueue());
+    char inputData1[] = {
+	1, 2, 3, 4, 5, 6, 7, 8};
+    char maxData[] = {4, 8};
+    char sumData[] = {10, 26};
+
+    std::vector<int> shape = {2,4,1}, shape2 = {2, 1}; // outer_size * channels * channel_size
+    webgpu::Tensor input(inputData1, shape, wgpu::BufferUsage::Storage, webgpu::Format::wFormatFp32);
+    webgpu::Tensor out(inputData1, shape, wgpu::BufferUsage::Storage | wgpu::BufferUsage::CopySrc, webgpu::Format::wFormatFp32);
+    std::vector<char *> blobs = {maxData, sumData};
+
+    webgpu::OpSoftmax op1(1, false);
+    op1.setBlobs(blobs, shape2);
+    op1.forward(input, out);
+    void * result;
+    out.mapReadAsync(result);
```

Comment:
Are you able to verify the value of the result?

Reply:
I am trying to manage the life cycle of Buffer, because Dawn/WebGPU only supports asynchronous callback function to read Buffer, which will be called after the whole program ends. The current problem is that the command buffer is submitted,  but the callback function was not called before the end of the program. . I am looking for a good way to read the buffer.

Reply:
You can simply add some waits before reading the result.

For example, it works for me.
```c
for(int i = 0; i < 10; ++i) {
  webgpu::wDevice->Tick();
  usleep(100);
}
```